atom/browser/api/atom_api_auto_updater.cc

```diff
@@ -62,16 +62,20 @@ void AutoUpdater::OnError(const std::string& message,
   v8::Locker locker(isolate());
   v8::HandleScope handle_scope(isolate());
   auto error = v8::Exception::Error(mate::StringToV8(isolate(), message));
-  auto errorObject = error->ToObject(
-    isolate()->GetCurrentContext()).ToLocalChecked();
 
   // add two new params for better error handling
-  errorObject->Set(mate::StringToV8(isolate(), "code"),
-                   v8::Integer::New(isolate(), code));
-  errorObject->Set(mate::StringToV8(isolate(), "domain"),
-                   mate::StringToV8(isolate(), domain));
-
-  mate::EmitEvent(isolate(), GetWrapper(), "error", errorObject, message);
+  ignore_result(v8::Object::Cast(*error)->DefineOwnProperty(
+        isolate()->GetCurrentContext(),
+        mate::StringToV8(isolate(), "code"),
+        v8::Integer::New(isolate(), code),
+        v8::PropertyAttribute::ReadOnly));
+  ignore_result(v8::Object::Cast(*error)->DefineOwnProperty(
+        isolate()->GetCurrentContext(),
+        mate::StringToV8(isolate(), "domain"),
+        mate::StringToV8(isolate(), domain),
+        v8::PropertyAttribute::ReadOnly));
+
+  mate::EmitEvent(isolate(), GetWrapper(), "error", error, message);
```

Comment:
I don't understand this fix. Is the crash fixed by making the properties immutable, or is it by passing error rather than errorObject? My _guess_ is the former and that the dropping errorObject as unnecessary is a bonus, but I'm not sure.

As an aside, if you have time for a short zoom 1-1 I'd like to talk over how you arrived at this fix -- I was looking at this crash as well and didn't zero in on code/domain as being the problem. I'd appreciate you walking through not just why this fixes it, but how you determined it, so that I can do a better job on my own fixes.